### ISSUE TITLE
make sure that alphas and Q2 grids agree in `alpha_s_bundle_pdf`

### ISSUE THREAD

**Issue by RoyStegeman** [NNPDF/nnpdf#2197]
The Q2grid of the alphas variation members in `NNPDF40_an3lo_as_01180_mhou_pdfas` is different from the Q2grid (and number of points in the alphas grid) in the .info file. This means that alphas cannot be computed for the error members. 

To avoid this in the future, it should be checked that these grids are consistent. 

The error can be reproduced as follows: 

```
>>> import lhapdf
>>> pdf = lhapdf.mkPDF("NNPDF40_an3lo_as_01180_mhou_pdfas/101")
>>> pdf.alphasQ2(10)
```
Which then raises the error  
```
RuntimeError: AlphaS value and Q interpolation arrays are differently sized
```

**Comment by scarlehoff** [comment]
What do you mean by that?

Is this related to this bug? https://github.com/NNPDF/eko/pull/391 (the ekos prepared before the fix was merged would produce a wrong-ish info file)

**Comment by RoyStegeman** [comment]
No, for `NNPDF40_an3lo_as_01180_mhou_pdfas` between mc and mb for members 101 and 102 the Q2 grid is 
1.5100000E+00 1.6500000E+00 1.7874390E+00 1.9429050E+00 2.1193750E+00 2.3204100E+00 2.5502940E+00 2.8142020E+00 3.1184120E+00 3.4705780E+00 3.8800750E+00 4.3584520E+00 4.9200000E+00

while for members 0-100 it is 
1.510000e+00 1.657384e+00 1.827949e+00 2.026319e+00 2.258232e+00 2.530851e+00 2.853170e+00 3.236569e+00 3.695538e+00 4.248669e+00 4.920000e+00

The `AlphaS_Qs` in the .info file is the same as the one for members 0-100, but obviously disagrees with what is there for members 101 and 102

Not sure why this happened originally, but it's something we can check for when combining multiple PDF sets into one.

**Comment by giacomomagni** [comment]
Thanks for the work @RoyStegeman, looks indeed much nicer.
 
I think one could also clean this and the following functions (although watch out the normalisation constants are different for the some equal variation length ):

https://github.com/NNPDF/nnpdf/blob/15d26d414fcfdac10ca6a4192d6b527fb0d2b4b6/validphys2/src/validphys/theorycovariance/construction.py#L276-L337
